src/OpenLoco/src/Viewport.cpp

```diff
@@ -216,7 +216,7 @@ namespace OpenLoco::Ui
         {
             auto orderRing = Vehicles::OrderRingView(orderFrame.orderOffset);
             auto* order = orderRing.atIndex(0);
-            if (!order->hasFlags(Vehicles::OrderFlags::HasNumber))
+            if (!order || !order->hasFlags(Vehicles::OrderFlags::HasNumber))
```

Comment:
I think this hides the issue rather than addressing it. Why is `order` a nullptr in the first place? The code assumes that there is an order to display, so why isn't there?

Reply:
Hi, yes I understand the problem a bit better now. I am still figuring out the issue.

Thank you

Reply:
Although it does hide the issue i tried to find the source before and wasn't successful. So if you can't work it out it doesn't matter we can just merge this

Reply:
Suggest we add a comment that we don't know why we can get into a bad state here but if we do then this makes it safe.

Reply:
Found the source of the issue. I'll keep this PR but augment it with an additional fix.